### ISSUE TITLE
Setting l2 is causing weird behavior of trained network

### ISSUE THREAD

**Issue by papperwing** [deeplearning4j/deeplearning4j#4250]
After setting L2 to any value network start behaving weirdly. Tested specifically on 1e-20 (that value I suppose should be also same as no l2 at all)
both Dropout and L1 regularization are set to 0

Network config: https://github.com/papperwing/image-net/blob/master/modules/image-net-model-creator/src/main/java/cz/muni/fi/image/net/model/creator/ModelBuilderImpl.java#L201-L267

**Comment by AlexDBlack** [comment]
After adapting your code (for future reference, it's *much* slower to debug these issues if I have to disentangle your code to make something workable first), I'm unable to reproduce this with random data after >800 iterations, on 0.9.1.

Can you give me something complete that I can run to reproduce this?

**Comment by papperwing** [comment]
Just pasting some stuff we exchanged on gitter
home of project: https://github.com/papperwing/image-net
dataset: https://gist.github.com/papperwing/dd91d455de5ae895b9568d54b0b7baf8
parameters of main I use: https://gist.github.com/papperwing/b7f4eefd099e8e0ea4fa221e750147fe

**Comment by papperwing** [comment]
Parameters which are causing me exploding activations and Alex has normal results:
https://gist.github.com/AlexDBlack/46776fae02dcebb7c13433513e863ea0

**Comment by AlexDBlack** [comment]
Right, this is what I'm getting with those parameters:

![image](https://user-images.githubusercontent.com/2360237/32438537-fe011244-c33e-11e7-91e1-bf35f4e44fd3.png)

**Comment by papperwing** [comment]
Same parameters as Alex computed on CPU:
![snimek obrazovky 16](https://user-images.githubusercontent.com/5627436/32439340-81480370-c2ee-11e7-979a-942c44a10244.png)

**Comment by papperwing** [comment]
After deleting whole project and new import it started working properly. I don't know what caused that.

**Comment by papperwing** [comment]
It showed up that because of weird git shenanigans the .regularization(true) didn't get into master. After deleting local repository and cloning it again from master it was missing. after adding that it behaves same as before. That mean that Alex didnt tried it as me.

**Comment by AlexDBlack** [comment]
After pulling the updated repo, I can confirm that I'm seeing the same behaviour with L2 of 1e-20, with sensible behaviour with L2=0.0.

**Comment by AlexDBlack** [comment]
Cause: l2 (from TransferLearning FineTuneConfiguration) was being incorrectly applied to frozen layers in ComputationGraph. Fixed here: https://github.com/deeplearning4j/deeplearning4j/pull/4260
